test/stress/cpp/microbench.cpp

```diff
@@ -0,0 +1,84 @@
+#include "test/jemalloc_test.h"
+#include "test/bench.h"
+
+static void
+malloc_free(void) {
+	void *p = malloc(1);
+	expect_ptr_not_null(p, "Unexpected new failure");
+	free(p);
+}
+
+static void
+new_delete(void) {
+	auto p = ::operator new(1);
+	expect_ptr_not_null(p, "Unexpected new failure");
+	::operator delete(p);
+}
+
+static void
+malloc_free_array(void) {
+	void *p = malloc(sizeof(int)*8);
+	expect_ptr_not_null(p, "Unexpected new[] failure");
+	free(p);
+}
+
+static void
+new_delete_array(void) {
+	auto p = new int[8];
+	expect_ptr_not_null(p, "Unexpected new[] failure");
+	delete[] p;
+}
+
+#if __cpp_sized_deallocation >= 201309
+static void
+new_sized_delete(void) {
+	auto p = ::operator new(1);
+	expect_ptr_not_null(p, "Unexpected new failure");
+	::operator delete(p, 1);
+}
+
+static void
+malloc_sdallocx(void) {
+	void *p = malloc(1);
+	expect_ptr_not_null(p, "Unexpected new failure");
+        sdallocx(p, 1, 0);
+}
+#endif
+
+TEST_BEGIN(test_free_vs_delete) {
+	compare_funcs(10*1000*1000, 100*1000*1000, 
+	    "malloc_free", (void *)malloc_free, 
+	    "new_delete", (void *)new_delete);
+}
+TEST_END
+
+TEST_BEGIN(test_free_array_vs_delete_array) {
+	compare_funcs(10*1000*1000, 100*1000*1000, 
+	    "malloc_free_array", (void *)malloc_free_array, 
+	    "delete_array", (void *)new_delete_array);
+}
+TEST_END
+
+#if __cpp_sized_deallocation >= 201309
```

Comment:
From this point below there are 3 `#if`s. We can merge them into one:
```
TEST_BEGIN(test_sized_delete_vs_sdallocx) {
#if __cpp_sized_deallocation >= 201309
    compare_funcs(...)
#else
    malloc_printf(...);
#endif
}
```

Reply:
Merged